### PR TITLE
Reports Tests for Public-API

### LINKED ISSUE #5: [public api] Create tests for /v2/report
Create tests for /v2/report with pytest

### PR DESCRIPTION
**Pull request by RusticPotatoes**
create tests for public-api reports https://github.com/Bot-detector/public-api/issues/5

### CODE REVIEW

File: src/app/models/report.py
Comment (extreme4all) [inline src/app/models/report.py:1]: what are you trying to achieve here?
it seems like you want to add warning logs, however these are duplicate because the parse_data() function does the logging

Comment (RusticPotatoes) [reply src/app/models/report.py:1]: this came into play when i was debugging how to get the model to accept none and an int, it wouldnt even get to the data parse method errors.  with the custom validator it should now.  these log messages can be removed.

File: src/app/views/input/report.py [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-    equip_head_id: int = Field(None, ge=0)
-    equip_amulet_id: int = Field(None, ge=0)
-    equip_torso_id: int = Field(None, ge=0)
-    equip_legs_id: int = Field(None, ge=0)
-    equip_boots_id: int = Field(None, ge=0)
-    equip_cape_id: int = Field(None, ge=0)
-    equip_hands_id: int = Field(None, ge=0)
-    equip_weapon_id: int = Field(None, ge=0)
-    equip_shield_id: int = Field(None, ge=0)
+    equip_head_id: Optional[int] = Field(0, ge=0)
```
Comment (extreme4all) [inline src/app/views/input/report.py]: Any reason why field is not default None

File: src/app/models/report.py
Comment (extreme4all) [reply src/app/models/report.py:1]: please revert the changes to the original file, the ternary operator is a bit cleaner

File: src/app/views/input/report.py (hunk 1 above)
Comment (RusticPotatoes) [reply src/app/views/input/report.py]: Cyborger1 said the game defaults the field to be 0 (for none).  optional[int] allows the equip_name_id to not be passed (none) but the model class will set default value of the item to 0 (none).

File: src/app/models/report.py
Comment (RusticPotatoes) [reply src/app/models/report.py:1]: > please revert the changes to the original file, the ternary operator is a bit cleaner

done

File: src/app/views/input/report.py (hunk 1 above)
Comment (extreme4all) [reply src/app/views/input/report.py]: if we set 0 here then on the db side we will get 0 instead of none if the key is not provided

Comment (RusticPotatoes) [reply src/app/views/input/report.py]: Okay I need to test adding no default (null or 0). I believe the optional[int] should cover this empty case but I need to verify when testing.

**Review by extreme4all — APPROVED** [review]
